src/nl/hannahsten/texifyidea/settings/conventions/TexifyConventionsGlobalSettingsManager.kt

```diff
@@ -0,0 +1,37 @@
+package nl.hannahsten.texifyidea.settings.conventions
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+
+/**
+ * Global convention settings.
+ *
+ * The global settings store a list of schemes, one of which is the global default scheme. In addition, the global
+ * settings remember which scheme is currently selected. Instances of this class must be serializable, since they are
+ * persisted by a [TexifyConventionsGlobalSettingsManager].
+ */
+data class TexifyConventionsGlobalState(
```

Comment:
So the idea is, that calls to Magic classes are replaced by calls to the scheme state (via the settings manager, the project one if you have it and otherwise global) to get the appropriate scheme?

Reply:
Yes, or more precisely: you should always use the `TexifyConventionsSettingsManager` since otherwise you won't know which scheme is active because you can't determine it from the `TexifyConventionsGlobalState` alone. The `TexifyConventionsSettingsManager` returns a `TexifyConventionsSettings` object which determines the active scheme (`currentScheme`) automatically. To do so, it needs both, a  `TexifyConventionsProjectState` and a `TexifyConventionsGlobalState` because for each project whether a project scheme is used or not is stored in the `TexifyConventionsProjectState`.

Reply:
Once the open questions are resolved I can port the label inspections to use the convention settings. That should make everything clearer.

Reply:
Makes sense to me, I don't think there are many cases where you would need access to conventions settings without knowing which project it is for.

Reply:
>  I can port the label inspections to use the convention settings

Yes, please do :)